packages/patternfly-3/patternfly-react-extensions/src/components/ClassificationBanner/ClassificationBanner.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+ * ClassificationBanner Component for PatternFly React
+ */
+class ClassificationBanner extends React.Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      closed: props.closed || false
+    }
+
+  }
+
+  onClose = ()=>{
+    this.setState({closed:true})
+  }
+
+  componentWillReceiveProps(nextProps) {
+    
+    if (nextProps.closed !== this.state.closed) {
+      this.setState({ closed: nextProps.closed});
+    }
+  }
+
+  render(){
+    const { 
+      children, 
+      bottomBanner,
+      hostName,
+      userName,
+      closeButton,
+      userNamePosition,
+      hostNamePosition,
+      bannerColor,
+      title,
+      ...props } = this.props;
+
+      const defaultColors = {
+        "pf-red": 1,
+        "pf-blue": 1,
+        "pf-green": 1
+      }
+
+    const classificationBannerClasses = {
+      top: classNames('classification-banner-pf-banner',
+      'classification-banner-pf-banner-top',
+      {
+        'classification-banner-pf-hide': this.state.closed,
+      },
+      defaultColors[bannerColor]?('classification-banner-'+bannerColor):''),
+      bottom: classNames(
+      'classification-banner-pf-banner',
+      defaultColors[bannerColor]?('classification-banner-'+bannerColor):'',
+      {
+      'classification-banner-pf-hide':(!bottomBanner)||this.state.closed
+      },
+      'classification-banner-pf-banner-bottom'),
+      closeButton: classNames({
+        'classification-banner-pf-close pficon-error-circle-o': closeButton
+      }),
+      children: classNames({
+        'classification-banner-pf-children-no-bottom':(!this.state.closed && !bottomBanner),
+        'classification-banner-pf-children':(!this.state.closed && bottomBanner)})
+    }
+
+    var leftLabels=[hostNamePosition==='left'?(<span>{hostName}</span>):null,
+    userNamePosition==='left'?(<span>{userName}</span>):null],
+    rightLabels=[hostNamePosition==='right'?(<span>{hostName}</span>):null,
+    userNamePosition==='right'?(<span>{userName}</span>):null];
+
+    var bannerBackgroundStyle = defaultColors[bannerColor]?{}:{'background':bannerColor};
+    
+    return(<div>
```

Comment:
```suggestion
    return(<div {...props}>
```
extra props should go on this div to allow setting of app specific parameters (id, classname)

Reply:
@jeff-phillips-18 done